classes/plugins/PKPPubIdPlugin.inc.php

```diff
@@ -0,0 +1,410 @@
+<?php
+
+/**
+ * @file classes/plugins/PKPPubIdPlugin.inc.php
+ *
+ * Copyright (c) 2014-2015 Simon Fraser University Library
+ * Copyright (c) 2003-2015 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class PKPPubIdPlugin
+ * @ingroup plugins
+ *
+ * @brief Abstract class for public identifiers plugins
+ */
+
+import('lib.pkp.classes.plugins.Plugin');
+
+abstract class PKPPubIdPlugin extends LazyLoadPlugin {
+
+	/**
+	 * Constructor
+	 */
+	function PKPPubIdPlugin() {
+		parent::LazyLoadPlugin();
+	}
+
+
+	//
+	// Implement template methods from Plugin
+	//
+	/**
+	 * @copydoc Plugin::register()
+	 */
+	function register($category, $path) {
+		if (!parent::register($category, $path)) return false;
+		// Enable storage of additional fields.
+		foreach($this->getDAOs() as $publicObjectType => $daoName) {
+			HookRegistry::register(strtolower_codesafe($daoName).'::getAdditionalFieldNames', array($this, 'getAdditionalFieldNames'));
+		}
+		return true;
+	}
+
+	/**
+	 * @copydoc Plugin::getActions()
+	 */
+	function getActions($request, $actionArgs) {
+		$router = $request->getRouter();
+		import('lib.pkp.classes.linkAction.request.AjaxModal');
+		return array_merge(
+				$this->getEnabled()?array(
+						new LinkAction(
+								'settings',
+								new AjaxModal(
+										$router->url($request, null, null, 'manage', null, $actionArgs),
+										$this->getDisplayName()
+										),
+								__('manager.plugins.settings'),
+								null
+								),
+				):array(),
+				parent::getActions($request, $actionArgs)
+				);
+	}
+
+ 	/**
+	 * @copydoc Plugin::manage()
+	 */
+	function manage($args, $request) {
+		$user = $request->getUser();
+		$router = $request->getRouter();
+		$context = $router->getContext($request);
+
+		$settingsFormName = $this->getSettingsFormName();
+		$settingsFormNameParts = explode('.', $settingsFormName);
+		$settingsFormClassName = array_pop($settingsFormNameParts);
+		$this->import($settingsFormName);
+		$form = new $settingsFormClassName($this, $context->getId());
+		$notificationManager = new NotificationManager();
+		if ($request->getUserVar('save')) {
```

Comment:
(See e.g. https://github.com/pkp/ojs/blob/master/plugins/generic/webFeed/WebFeedPlugin.inc.php which is a little different than what I quoted above)